### PR TITLE
feat(organizations): add endpoints to handle organization members TASK-963

### PR DESCRIPTION
**Pull request by rajpatel24**
### 🗒️ Checklist

1. [x] run linter locally
2. [x] update all related docs (API, README, inline, etc.), if any
3. [x] draft PR with a title `<type>(<scope>)<!>: <title> TASK-1234`
4. [x] tag PR: at least `frontend` or `backend` unless it's global
5. [x] fill in the template below and delete template comments
6. [x] review thyself: read the diff and repro the preview as written
7. [x] open PR & confirm that CI passes
8. [ ] request reviewers, if needed
9. [ ] delete this section before merging

### 📣 Summary
- Enhance the organization member management API to support role updates and member removal via PATCH and DELETE requests, excluding member creation.

### 📖 Description
- This update introduces endpoints for managing organization members:

    - GET `/api/v2/organizations/<org-id>/members/` to list all members in an organization.
    - PATCH `/api/v2/organizations/<org-id>/members/<username>/` to update a member's role (e.g., promote to admin).
    - DELETE `/api/v2/organizations/<org-id>/members/<username>/` to remove a member from the organization.

Note: Creating members is not supported via this endpoint. Roles are restricted to 'admin' or 'member'. Including the details of invited members (those who have not yet joined the organization) is not covered in this update.

### 👷 Description for instance maintainers
- This update provides new functionality to manage organization members:

    - Role updates for members (promotion/demotion between 'admin' and 'member').
    - Member removal from an organization.
    - The endpoints are optimized to use database joins to fetch member roles efficiently without excessive database queries, ensuring minimal load.

### 👀 Preview steps
1. ℹ️ Have an account and an organization with multiple members.
2. Use the GET method to list the members of the organization.
3. Use the PATCH method to update a member's role (e.g., promote a user to admin).
4. Use the DELETE method to remove a member from the organization.
5. 🟢 Notice the endpoints behave as expected, with valid role updates and member removal.

### 💭 Notes
- The code uses database joins to optimize role determination for members, avoiding inefficient role lookups.
- Role validation restricts role assignments to 'admin' or 'member' only.
- Including invited members' details is not covered in this update. This will be implemented in a future update.

### CODE REVIEW

File: kpi/paginators.py
Comment (noliveleger) [inline kpi/paginators.py:147]: I would rename the class just to match what it is paginating (i.e.: organization members)

File: kpi/paginators.py
Comment (noliveleger) [inline kpi/paginators.py:151]: Is `10` a requirement?

File: kobo/apps/organizations/serializers.py
Comment (noliveleger) [inline kobo/apps/organizations/serializers.py:49]: Actually, you need to get if the user has MFA enabled. (e.g. `MfaMethod.objects.filter(user=obj, is_active=True).exists()`). Ensure to optimize this for a list like you did for role ;-)

File: kobo/apps/organizations/tests/test_organization_members_api.py
Comment (noliveleger) [inline kobo/apps/organizations/tests/test_organization_members_api.py:16]: For future-proof validaiton, please add `mmo_override=True` 
This can be simplied with

```python
self.organization = baker.make(Organization, id='org_12345', mmo_override=True)
self.owner_user = baker.make(User, username='owner')
self.member_user = baker.make(User, username='member')
self.invited_user = baker.make(User, username='invited')

self.organization.add_user(self.owner_user)  # first user added to org, is always the owner and an admin
self.organization.add_user(self.member_user, is_admin=False) 
```

File: kobo/apps/organizations/tests/test_organization_members_api.py
Comment (noliveleger) [inline kobo/apps/organizations/tests/test_organization_members_api.py:19]: `invited_user` is not used at all for the moment, right? Let's remove until we implement invitations.

File: kobo/apps/organizations/tests/test_organization_members_api.py
Comment (noliveleger) [inline kobo/apps/organizations/tests/test_organization_members_api.py:75]: All your tests are done with `self.client == self.owner` , but we need to validate permissions for regular members, admins and anonymous. Let's those tests as soon as #5218 is merged. I will tag this PR `blocked by`

File: kobo/apps/organizations/serializers.py
Comment (noliveleger) [inline kobo/apps/organizations/serializers.py:32]: I think we should use `user__is_active` to keep the same convention as other fields.

File: kobo/apps/organizations/serializers.py
Comment (noliveleger) [inline kobo/apps/organizations/serializers.py:24]: I think we should use `user__has_mfa_enabled` to keep the same convention as other fields.

**Review by noliveleger — CHANGES_REQUESTED** [review]

File: kobo/apps/organizations/views.py
Comment (noliveleger) [inline kobo/apps/organizations/views.py:374]: Let's move the logic to the serializer to avoid calling the "update" twice. something like

```python 
def update(self, instance, validated_data):
     if role := validated_data.pop('role', None)
         validated_data['is_admin'] = role == 'admin'
     return super().update(instance, validated_data)
```

**Review by noliveleger — CHANGES_REQUESTED** [review]

**Comment by noliveleger** [comment]
This PR is blocked by #5218

**Comment by magicznyleszek** [comment]
@noliveleger @rajpatel24 this is no longer blocked, right?

File: kpi/paginators.py
Comment (rajpatel24) [reply kpi/paginators.py:151]: 10 is not a requirement defined in the task. I need to check with you or the team to determine the appropriate value for it.

File: kobo/apps/organizations/tests/test_organization_members_api.py
Comment (rajpatel24) [reply kobo/apps/organizations/tests/test_organization_members_api.py]: Thanks for pointing that out! I've added the role-based validation tests for organization member permissions

File: kobo/apps/organizations/serializers.py
Comment (rajpatel24) [reply kobo/apps/organizations/serializers.py:17]: Achieved this using `http_method_names`. Added a test for it as well

File: kobo/apps/organizations/views.py
Comment (rajpatel24) [reply kobo/apps/organizations/views.py]: Done! I made a slight adjustment to your suggested implementation. I opted to use `get` instead of `pop` when accessing the `role` value from `validated_data`, as this allows the updated value to be properly reflected in the API response :-)

File: kpi/paginators.py
Comment (jamesrkiger) [reply kpi/paginators.py]: @rajpatel24 Can we change this to limit offset pagination? Our table components on the frontend currently assume limit offset pagination and @noliveleger said it shouldn't cause any problems to use that here.

File: kpi/paginators.py
Comment (noliveleger) [inline kpi/paginators.py:146]: You can delete this class, not needed anymore

File: kobo/apps/organizations/serializers.py
Comment (noliveleger) [inline kobo/apps/organizations/serializers.py:32]: I think that's the wrong field. You need to retrieve  `User.extra_details.data['name']` . BTW in that case, the field should be `user__extra_details__name` . 

cc @jamesrkiger

File: kobo/apps/organizations/views.py
Comment (noliveleger) [inline kobo/apps/organizations/views.py:90]: Asked by the front-end team, let's use `LimitOffsetPagination` instead.

**Review by noliveleger — CHANGES_REQUESTED** [review]

File: kobo/apps/organizations/views.py
Comment (noliveleger) [inline kobo/apps/organizations/views.py:466]: You can remove this method entirely. Right now, we want to delete the `OrganizationUser` only.  Do not touch the `User` object.

File: kpi/paginators.py
Comment (rajpatel24) [reply kpi/paginators.py]: @jamesrkiger I have removed the custom pagination class for the organization members API. It will now use `LimitOffsetPagination` by default.

File: kpi/paginators.py
Comment (rajpatel24) [reply kpi/paginators.py]: Done. It will use `LimitOffsetPagination` by default

**Review by noliveleger — APPROVED** [review]
LGTM 

For the record: The complex query took only few millisecond on a 500 members org.

**Pull request merged**
